### DISCUSSION TITLE
Any guidance on r_max with regard to an input set?

### DISCUSSION THREAD

**Discussion by vputz** [mir-group/allegro#52] [Q&A] [answered]
It's clear to me that setting r_max must have pretty significant impacts on the resultant model (if nothing else, raising it seems to increase GPU memory usage!), and this makes sense as a few things seem to depend on the available nearest neighbors for each in put atom.

But is there any guidance on setting r_max?  For the input aspirin MD17 (aspirin_ccsd-train.npz) the given `r_max` of 6.0 ensured 10 nearest neighbors, if I did the calculations correctly.  Should we think of `r_max` in those terms ("choosing r_max such that each atom has X nearest neighbors" or some other heuristic?) or are there any other guidelines?  I couldn't find much searching here or nequip.

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @vputz ,

Thanks for your interest in our work!

Right, `r_max` is one of the most important hyperparameters for Allegro, and can have significant impact on performance (linear scaling) as you noticed.

Generally, `r_max` must be large enough to capture the immediate "first shell" of neighbors, and depending on your system and requirements you may see gains from moving out to the "second shell" and so on. You can gain some sense of this for your system by looking at the RDFs of your training data or structures you want to be able to run inference on.
